lib/glific/contacts/import.ex

```diff
@@ -184,6 +132,78 @@ defmodule Glific.Contacts.Import do
     end
   end
 
+
+
+  @spec process_data(User.t(), map(), non_neg_integer) :: Contact.t() | map()
+  def process_data(_user, %{delete: "1"} = contact, _contact_attrs) do
+    Repo.put_process_state(contact.organization_id)
+    case Repo.get_by(Contact, %{phone: contact.phone}) do
+      nil ->
+        %{ok: "Contact does not exist"}
+
+      contact ->
+       {:ok, contact} = Contacts.delete_contact(contact)
+       contact
+    end
+  end
+
+  def process_data(user, contact_attrs, group_id) do
+    Repo.put_process_state(contact_attrs.organization_id)
```

Comment:
Let's move this Repo.put_process_state(contact_attrs.organization_id) thing in async funcation instead of repeating it every where.

Reply:
which async function?